base/compiler/abstractinterpretation.jl

```diff
@@ -407,6 +424,7 @@ function abstract_call_method(interp::AbstractInterpreter, method::Method, @nosp
             # continue inference, but note that we've limited parameter complexity
             # on this call (to ensure convergence), so that we don't cache this result
             if call_result_unused(sv)
+                add_remark!(interp, sv, "Widening unused call result early to avoid unbounded recursion")
```

Comment:
This wording is misleading: there's never widening that happens here (except if the code-path would be discovered to stack overflow instead), since the result is known to be unused, though we also don't know what will happen next (though it's unlikely inlining would succeed regardless due to the graph cycle).
```suggestion
                add_remark!(interp, sv, "Delaying call result analysis until after optimization due to unbounded recursion")
```